### PR TITLE
feat(parser): enhance pipeline parsing to handle quoted arrows

### PR DESCRIPTION
**Pull request by glav**
- Introduced quote-aware parsing to distinguish between pipeline operator syntax (`-> @agent`) and casual mentions of `->` in user messages.
- Added helper functions to check if text is within quotes and to split pipeline commands while respecting quoted content.
- Updated the REPL parser to utilize the new quote-aware logic, ensuring that quoted arrows do not trigger pipeline parsing.
- Implemented acceptance tests to validate the new behavior, ensuring that quoted arrows are handled correctly and do not interfere with valid pipeline commands.
- Enhanced existing tests to cover edge cases involving nested quotes and mixed quoted/unquoted arrows.

### CODE REVIEW

File: tests/test_pipeline_parse_error_acceptance.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        """
+        # Call REAL implementation
+        result = parse_command("@pm explain how the '->' operator works in TeamBot")
+
+        # Verify: result.is_pipeline == False and '->' appears in result.content
+        assert result.is_pipeline is False, "Should NOT be parsed as pipeline"
+        assert result.agent_id == "pm", "Should route to PM agent"
+        assert "'->'".replace("'", "'") in result.content or "'->" in result.content, (
+            "Content should include the quoted arrow operator"
+        )
```
[the remark this answers is not in the document]
Comment (glav) [reply tests/test_pipeline_parse_error_acceptance.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/glav/teambot/pull/189#discussion_r2881646590)

File: tests/test_pipeline_parse_error_acceptance.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        """
+        # Call REAL implementation
+        result = parse_command("@pm the syntax is \"use '->' between agents\"")
+
+        # Verify: is_pipeline == False, full quoted string preserved in content
+        assert result.is_pipeline is False, "Should NOT be parsed as pipeline"
+        assert result.agent_id == "pm", "Should route to PM agent"
+        assert "'->'".replace("'", "'") in result.content or "'->" in result.content, (
+            "Content should preserve the nested quoted arrow"
+        )
```
[the remark this answers is not in the document]
Comment (glav) [reply tests/test_pipeline_parse_error_acceptance.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/glav/teambot/pull/189#discussion_r2881646601)
